apps/core/components/searchForm/SearchFormModes.js

```diff
@@ -75,6 +95,25 @@ class SearchFormModes extends React.Component<Props> {
           infants={infants}
           bags={bags}
         />
+        <Modal isVisible={this.state.isModalOpen} onClose={this.onClose}>
+          {this.state.formMode === FORM_MODE.TRIP_TYPE && (
+            <Select
+              optionsData={TRIP_TYPE}
+              selectedType={tripType}
+              onSelect={this.handleTripTypeSelect}
+              onClosePress={this.onClose}
+            />
+          )}
```

Comment:
Please also refactor this part. One modal for trip type and one modal for passengers input.